python/tvm/relay/transform.py

```diff
@@ -1032,3 +1032,40 @@ def visit_var(self, var):
                 else:
                     return var
         return ChangeBatchMutator().visit(func)
+
+def add_dispatch_func(mod, func_name, input_shape, dispatch_func):
```

Comment:
I think we can make it a `ModulePass` can func_name can default to "main" so that it can be managed by pass manager. The API would look like:

```python
AddDispatchFunc(input_shape, dispatch_func, func_name="main")
```